### PR TITLE
api: add email verification lookup and rework verify endpoint

### PR DESCRIPTION
**Pull request by dcow**
Add lookup table to get verification status by email.

### CODE REVIEW

File: api/src/lib.rs [hunk 1]
```diff
@@ -1545,15 +1578,19 @@ where
             tide::with_state(State::new(verify_db.clone(), token_db.clone()));
 
         verify_tokens
-            .at(":id")
+            .at("tokens/:id")
```
Comment (dcow) [inline api/src/lib.rs:1581]: s/tokens/entries

**Comment by dcow** [comment]
@ceberly can you point me to the js that needs to be updated to incorporate the new routes?

File: api/src/lib.rs [hunk 2]
```diff
@@ -1540,20 +1606,50 @@ where
...
+        // Legacy verification endpoint. Remove for API v3
+        let mut verify_tokens_legacy =
             tide::with_state(State::new(verify_db.clone(), token_db.clone()));
 
-        verify_tokens
+        verify_tokens_legacy
             .at(":id")
             .with(ensure_vault_id)
-            .with(cors)
+            .with(signed_pow_auth)
```
Comment (dcow) [inline api/src/lib.rs:1616]: @ceberly we need to adjust this to make sure the PUT is not behind auth, right? (re: the convo on 93)

Comment (dcow) [reply api/src/lib.rs:1613]: see [78b7bc8](https://github.com/withuno/identity/pull/92/commits/78b7bc8457bb26bd5e3f0ad2f62ad723019464f2)

**Review by ceberly — APPROVED** [review]

**Pull request merged**